src/components/Recipe/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { createPortal } from 'react-dom'
+
+import { fetchRecipe } from 'api'
+import Loader from 'components/common/Loader'
+import { Ingredient, RecipeDetails, Step } from 'types'
+
+import styles from './Recipe.module.scss'
+
+function Recipe() {
+  const { id } = useParams()
+
+  const [recipe, setRecipe] = useState<RecipeDetails>()
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    fetchRecipe(id!).then((response) => {
+      setRecipe(response.data)
+      setIsLoading(false)
+    })
+  }, [id])
+
+  return (
+    <>
+      {isLoading && createPortal(<Loader />, document.body)}
+      {recipe && (
+        <div className={styles.container}>
+          <h2 className={styles.title}>{recipe.title}</h2>
+          <img className={styles.photo} src={recipe.image} alt="" />
+          <div className={styles.details}>
+            <div className={styles.ingredients}>
+              <h3 className={styles.ingredients__title}>Ingredients</h3>
+              <ul className={styles.ingredients__list}>
+                {recipe.extendedIngredients.map((ingredient: Ingredient) => {
+                  return (
+                    <li
+                      className={styles.ingredients__item}
+                      key={ingredient.original}
+                    >
+                      {ingredient.original}
+                    </li>
+                  )
+                })}
+              </ul>
+            </div>
+            <div className={styles.steps}>
+              <h3 className={styles.steps__title}>Steps</h3>
+              <ol className={styles.steps__list}>
+                {recipe.analyzedInstructions[0].steps.map((step: Step) => {
```

Comment:
what about next elements in array? because you're only using [0]

Reply:
@sasza2 sooo this is how API I use is designed. It always returns an array with one element (object) and inside this object there is another array with steps I want to loop through.

Example:
```
analyzedInstructions: [
  {
    name: '',
    steps: [
      {
        number: 1,
        step: 'Spread pineapple in bottom of a 9X13 pan',
        ingredients: [
          {
            id: 9266,
            name: 'pineapple',
            localizedName: 'pineapple',
            image: 'pineapple.jpg',
          },
          {
            id: 0,
            name: 'spread',
            localizedName: 'spread',
            image: '',
          },
        ],
      },
      {
        number: 2,
        step: 'Cover with pie filling',
        ingredients: [
          {
            id: 0,
            name: 'pie filling',
            localizedName: 'pie filling',
            image: '',
          },
        ],
      },
    ],
  },
]